iothub/device/tests/DeviceClientTests.cs

```diff
@@ -1773,9 +1775,291 @@ public void DeviceClient_InitWithHttpTransportButNoModelId_DoesNotThrow()
                 .NotThrow();
         }
 
+
+        [Description("Testing current behavior, The logically correct expected exception should be OperationCancelledException but current expected exception is IotHubCommunicationException since we try to avoid behavior change.")]
+        [TestMethod]
+        [ExpectedException(typeof(IotHubCommunicationException))]
```

Comment:
Consider using FluentAssertions instead. You can validate a method throws.

https://fluentassertions.com/exceptions/

Reply:
Fixing the whole file.